gateway-ha/src/test/java/io/trino/gateway/ha/TestGatewayHaMultipleBackend.java

```diff
@@ -359,6 +359,23 @@ void testCookieSigning()
         assertThat(callbackResponse.code()).isEqualTo(500);
     }
 
+    @Test
+    void testHealthCheckEndpoints()
+            throws IOException
+    {
+        Request livenessCheck = new Request.Builder()
+                        .url("http://localhost:" + routerPort + "/trino-gateway/livez")
+                        .build();
+        Response livenessResponse = httpClient.newCall(livenessCheck).execute();
+        assertThat(livenessResponse.code()).isEqualTo(200);
+
+        Request readinessCheck = new Request.Builder()
+                .url("http://localhost:" + routerPort + "/trino-gateway/readyz")
+                .build();
+        Response readinessResponse = httpClient.newCall(readinessCheck).execute();
```

Comment:
Is there any risk of this test being flaky? Should it try multiple times?

Reply:
MO there should be a loop with some very small sleep (100ms).

Reply:
Updated.